src/app/core/audio/player.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subscriber } from 'rxjs';
+
+export interface AudioMetadata {
+  duration: number;
+  format: string;
+  bitrate: number;
+  frequency: number;
+}
+
+export interface PlaybackMetadata {
+  duration: number;
+  progress: number;
+}
+
+@Injectable()
+export class PlayerService {
+
+  private playing: Subscriber<PlaybackMetadata>;
+  private playingPlayer: AV.Player;
+  private playingData: PlaybackMetadata;
```

Comment:
A bit wonky, but I added these "currently playing" singletons to support the static `stop()` and `seek()` methods on this service.